File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    boat: int = 0
    people.sort()
    i: int = 0
    j: int = len(people) - 1
    
    while i <= j:
        if limit - people[j] >= people[i]:
            i += 1
        j -= 1
        boat += 1
    return boat

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int len = people.length;
        int i, j;
        i = 0;
        j = len-1;
        Arrays.sort(people);
        
        while (i <= j) {
            int min = people[i];
            int max = people[j];
            if ((min + max) <= limit)
                i++;
            j--;
            answer += 1;
        }
        return answer;
    }
}